saleor/graphql/product/mutations/products.py

```diff
@@ -846,3 +851,161 @@ def mutate(cls, root, info, image_id, variant_id):
                     variant_image.delete()
         return VariantImageUnassign(
             product_variant=variant, image=image, errors=errors)
+
+class ProductVariantDigitalInput(graphene.InputObjectType):
+    max_downloads = graphene.Int(
+        description='[Optional] Maximal number of downloading for one url',
+        required=False, name='max_downloads')
+    url_valid_days = graphene.Int(
+        description='[Optional] Number of days which each url will be active',
+        required=False, name='url_valid_days'
+    )
+    automatic_fulfillment = graphene.Boolean(
+        description='Overwrite default automatic_fulfillment setting for variant',
+        required=False, name='automatic_fulfillment'
+    )
+
+
+class ProductVariantDigitalUploadInput(ProductVariantDigitalInput):
+    content_file = Upload(
+        required=True,
+        description='Represents an file in a multipart request.')
+
+
+class ProductVariantDigitalUpload(BaseMutation):
```

Comment:
And here I would use the name `DigitalContentCreate` - should be quite simple and easy to understand what it does just by looking at the schema.

Reply:
Have you tried using `ModelMutation` here?

Reply:
I checked `ModelMutation` previously but I wasn't sure what I can achieve by using it. As you see I have to overwrite `clean_input` method as well as `mutate` method. AFAIK `ModelMutation` will not give me anything